eclair-core/src/main/scala/fr/acinq/eclair/router/Validation.scala

```diff
@@ -257,55 +261,61 @@ object Validation {
     }
   }
 
+  private def applyChannelUpdate(pc: PublicChannel, update: Either[LocalChannelUpdate, RemoteChannelUpdate]): PublicChannel = update match {
+    case Left(lcu) => pc.updateChannelUpdateSameSideAs(lcu.channelUpdate).updateBalances(lcu.commitments)
+    case Right(rcu) => pc.updateChannelUpdateSameSideAs(rcu.channelUpdate)
+  }
+
+  private def applyChannelUpdate(pc: PrivateChannel, update: Either[LocalChannelUpdate, RemoteChannelUpdate]): PrivateChannel = update match {
+    case Left(lcu) => pc.updateChannelUpdateSameSideAs(lcu.channelUpdate).updateBalances(lcu.commitments)
+    case Right(rcu) => pc.updateChannelUpdateSameSideAs(rcu.channelUpdate)
+  }
+
```

Comment:
Shouldn't those be part of `PublicChannel`/`PrivateChannel`?

Reply:
I hesitated, we can.